Geometry/VeryForwardGeometryBuilder/interface/DetGeomDesc.h

```diff
@@ -30,62 +37,111 @@ class CTPPSRPAlignmentCorrectionData;
  \verbatim
     x_g = rotation * x_l + translation
  \endverbatim
+ *
+ * July 2020: Migrated to DD4hep
+ * To avoid any regression with values from XMLs / Geant4, all lengths are converted from cm (DD4hep) to mm. 
+ *
  **/
 
+struct DiamondDimensions {
+  double xHalfWidth;
+  double yHalfWidth;
+  double zHalfWidth;
+};
+
 class DetGeomDesc {
 public:
   using Container = std::vector<DetGeomDesc*>;
   using RotationMatrix = ROOT::Math::Rotation3D;
   using Translation = ROOT::Math::DisplacementVector3D<ROOT::Math::Cartesian3D<double>>;
 
-  ///Constructors to be used when looping over DDD
-  DetGeomDesc(DDFilteredView* fv);
+  // Constructor from old DD DDFilteredView
+  DetGeomDesc(const DDFilteredView& fv);
+  // Constructor from DD4Hep DDFilteredView
+  DetGeomDesc(const cms::DDFilteredView& fv, const cms::DDSpecParRegistry& allSpecParSections);
 
-  /// copy constructor and assignment operator
   DetGeomDesc(const DetGeomDesc&);
   DetGeomDesc& operator=(const DetGeomDesc&);
-
-  /// destructor
   virtual ~DetGeomDesc();
 
-  /// ID stuff
-  void setGeographicalID(DetId id) { m_geographicalID = id; }
-  DetId geographicalID() const { return m_geographicalID; }
+  // general info
+  const std::string& name() const { return m_name; }
+  int copyno() const { return m_copy; }
 
-  /// access to the tree structure
-  Container components() const;
-  float parentZPosition() const { return m_z; }
+  // is DD4hep
+  bool isDD4hep() const { return m_isDD4hep; }
+
+  // placement info
+  const Translation& translation() const { return m_trans; }  // in mm
+  const RotationMatrix& rotation() const { return m_rot; }
+
+  // shape info
+  // params() is left for general access to solid shape parameters, but should be used
+  // only with great care, for two reasons: 1. order of parameters may possibly change from
+  // a version to another of DD4hep; 2. length parameters unit is cm while PPS uses mm.
+  const std::vector<double>& params() const { return m_params; }  // default unit: mm from oldDD, cm from DD4hep
```

Comment:
@davidlange6 The params() member function in the public interface returns const std::vector<double>& though (it returns the member data m_params).